torchvision/models/regnet.py

```diff
@@ -0,0 +1,785 @@
+# Modified from
+# https://github.com/facebookresearch/ClassyVision/blob/main/classy_vision/models/anynet.py
+# https://github.com/facebookresearch/ClassyVision/blob/main/classy_vision/models/regnet.py
+
+
+import numpy as np
+import math
+import torch
+
+from collections import OrderedDict
+from enum import Enum, auto
+from typing import Any, Callable, List, Optional, Tuple
+from torch import nn, Tensor
+
+from .._internally_replaced_utils import load_state_dict_from_url
+from torchvision.models.mobilenetv2 import _make_divisible
+
+model_urls = {
+    # TODO(kazhang): add pretrained weights
+    "regnet_y_400m": "",
+}
+
+
+class _SqueezeExcitation(nn.Module):
+    """
+    Squeeze and excitation layer from
+    `"Squeeze-and-Excitation Networks" <https://arxiv.org/pdf/1709.01507>`_.
+    """
+
+    def __init__(
+        self,
+        in_channels: int,
+        reduction_ratio: Optional[int] = 16,
+        reduced_channels: Optional[int] = None,
```

Comment:
I agree that we should reduce the code duplication. In fact, we're not using `reductiton_ratio` in the model implementation. After removing it, the only difference between this and the one in [efficentnet](https://github.com/pytorch/vision/blob/main/torchvision/models/efficientnet.py#L34)  seems to be the activation. Maybe we could make the efficientnet one extendible. The question is should we do it before or after this PR :)

Reply:
Since you agree, I would rather if we do it earlier than later (because of the upcoming release). I also like your idea of marking the class with `_` to indicate it's private, let's do that on efficientnet (though I would like to bring up in our discussion the developer API). Could you send a PR where you extend the efficientnet class to support different activations so that you could reuse it?